### PR TITLE
Reenable GatewayTLSOrigination Test

### PR DESCRIPTION
**Pull request by nschhina**
Test for CI stability

### CODE REVIEW

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go [hunk 1]
```diff
@@ -98,46 +102,41 @@ func TestEgressGatewayTls(t *testing.T) {
...
 			for name, tc := range testCases {
 				t.Run(name, func(t *testing.T) {
-					bufDestinationRule := createDestinationRule(t, appNamespace, tc.destinationRuleMode, tc.fakeRootCert)
+					bufDestinationRule := createDestinationRule(t, serviceNamespace, tc.destinationRuleMode, tc.fakeRootCert)
+
+					istioCfg := istio.DefaultConfigOrFail(t, ctx)
+					systemNamespace := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+
+					// Wait for all CDS configs to update
+					time.Sleep(10 * time.Second)
```
Comment (JimmyCYJ) [inline tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:126]: Is there any way to check if CDS updates are finished?

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go [hunk 2]
```diff
@@ -98,46 +102,41 @@ func TestEgressGatewayTls(t *testing.T) {
...
 
-					ctx.Config().ApplyYAMLOrFail(ctx, appNamespace.Name(), bufDestinationRule.String())
-					defer ctx.Config().DeleteYAMLOrFail(ctx, appNamespace.Name(), bufDestinationRule.String())
+					ctx.Config().ApplyYAMLOrFail(ctx, systemNamespace.Name(), bufDestinationRule.String())
+					defer ctx.Config().DeleteYAMLOrFail(ctx, systemNamespace.Name(), bufDestinationRule.String())
+
+					// Wait for all CDS configs to update
+					time.Sleep(10 * time.Second)
```
Comment (JimmyCYJ) [inline tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:132]: same here. 20 secs wait in a test is not good. It is better to keeping checking updates periodically and return early.

**Comment by nschhina** [comment]
closes #25069

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go [hunk 3]
```diff
@@ -305,11 +307,6 @@ func setupEcho(t *testing.T, ctx resource.Context) (echo.Instance, echo.Instance
 		}).
 		BuildOrFail(t)
 
-	// Create a sidecarScope to only allow traffic to service namespace so that the traffic only goes
-	// through the egress gateway set up in service namespace. This way client cannot directly call our server
-	// in the same namespace("app") and is forced to route traffic to egress gateway
-	createSidecarScope(t, ctx, appsNamespace, serviceNamespace)
```
Comment (JimmyCYJ) [inline tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:311]: Why is this removed?

Comment (nschhina) [reply tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:311]: Since we are going from {client Namespace} -> {istio-system} -> {server namespace}  we don't need sidecar scope anymore. Now the server is deployed in a completely different namespace

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go (hunk 2 above)
Comment (nschhina) [reply tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go]: I reduced this wait time, only issue to periodically checking is how to get gateway config dump to check against dynamicactiveclusters. I think new wait time is reasonable if we find no more flakes.

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go (hunk 1 above)
Comment (nschhina) [reply tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go]: only issue to periodically checking is how to get gateway config dump to check against dynamicactiveclusters. I think new wait time is reasonable if we find no more flakes.

File: tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go [hunk 4]
```diff
@@ -370,15 +327,25 @@ spec:
...
-        - some-external-site.com
+        - server.{{.ServerNamespace}}.svc.cluster.local
 ---
 apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: egressgateway-for-server
+spec:
+  host: istio-egressgateway.istio-system.svc.cluster.local
+  subsets:
```
Comment (howardjohn) [inline tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:338]: why do we need this subset?

Comment (nschhina) [reply tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:338]: just being consistent with istio.io docs which uses subsets

Comment (nschhina) [reply tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go:338]: Also I would say having a subset is strictly more rigorous test for CDS generation
